app/models/user.rb

```diff
@@ -1,4 +1,7 @@
 class User < ApplicationRecord
+  has_secure_password
   has_many :reservations
   has_many :hotels, through: :reservations
+
+  validates :email, presence: true, uniqueness: true
 end
```

Comment:
- [ ] **Required!** In my opinion **uniqueness: true** should be for "username" or maybe for both username and email and kindly implement validations for **username** and **password**
     Maybe just like this or as you want.
![image](https://user-images.githubusercontent.com/65109017/211633908-1072ed5a-bded-4200-8cb6-2aa5b17f696f.png)